Classes/Emogrifier.php

```diff
@@ -924,21 +924,26 @@ private function copyCssWithMediaToStyleNode(\DOMDocument $xmlDocument, \DOMXPat
 
     /**
      * Extracts the media queries from $css.
+     * Skips empty media queries.
+     * @link http://stackoverflow.com/questions/36910664/recursive-subroutine-regex-to-match-css-media-queries/36911319#36911319
      *
      * @param string $css
      *
      * @return string[][] numeric array with string sub-arrays with the keys "css" and "query"
      */
     private function extractMediaQueriesFromCss($css)
     {
-        preg_match_all('#(?<query>@media[^{]*\\{(?<css>(.*?)\\})(\\s*)\\})#s', $css, $mediaQueries);
+        preg_match_all('/@media\b[^{]*({((?:[^{}]+|(?1))*)})/', $css, $mediaQueries, PREG_SET_ORDER);
         $result = [];
-        foreach (array_keys($mediaQueries['css']) as $key) {
-            $result[] = [
-                'css' => $mediaQueries['css'][$key],
-                'query' => $mediaQueries['query'][$key],
-            ];
+        foreach ($mediaQueries as $query) {
+            if (!empty(trim($query[2]))) {
```

Comment:
You'll need to put trim(...) into a local variable so the empty call also works in PHP 5.4:
https://travis-ci.org/jjriv/emogrifier/jobs/128679215#L181